doc/specs/#147 - Support different release channels [released, beta, alpha].md

```diff
@@ -0,0 +1,259 @@
+---
+author: Kaleb Luedtke - Trenly <trenlymc@gmail.com>
+created on: 2021-11-18
+last updated: 2021-11-20
+issue id: 147
+---
+
+# Release Channels
+
+For [#147](https://github.com/microsoft/winget-cli/issues/147)
+
+## Abstract
+
+Several packages have different channels for releasing different versions of their software. The Windows Package Manager should be able to support packages of the same Package Identifier where the versions come from different release channels. The Windows Package Manager should be also able to inform a user about which channels are available.
+
+## Inspiration
+
+Some other package managers have this capability. In the current state, a new package identifier must be created for each release channel which can cause conflicts when the versions are *not* side-by-side compatible.
+
+## Solution Design
+
+### Manifest Structure
+The Windows Package Manager manifest v1.1.0 schema has provided a key for declaring which channel an installer belongs to. However, this has the potential to make the installer manifests extremely complex when there are multiple release channels with multiple installers each.
+> The `Channel` key should be removed from the manifest v1.1.0 schema.
+
+Instead, the use of multiple installer manifest files will be used to define and manage the various channels available for a package in a way similar to locales. This will ensure that the client remains backwards compatible with old manifest versions, and will make the creation and maintenance of the various channels easier for contributors in the community repository. This will require a key be added to the  manifest schema. Channels should not be compatible with singleton manifests.
```

Comment:
Everything needs be compatible with any manifest representation, we just impose certain patterns on `winget-pkgs`.

Reply:
I'm not entirely sure what you mean by this comment. Is this just saying that singletons should support channels?

Reply:
Manifests from any source, i.e. REST sources that don’t use the manifest schema, preindexed sources (which are generated from the YAML files but converted to a different format), and local (-m on a folder) manifests is what I think that means.

Reply:
I see; In that case, there still shouldn’t be an issue, since the rest sources should still be able to specify the channel, pre-indexed should work as it would just require them to support building from the proposed mutiple-installer-manifests yaml structure, and local would still be handled just fine since the CLI would be able to find the channels based on the filenames or the specified value in the manifest file